pkg/scd/store/cockroach/availability.go

```diff
@@ -0,0 +1,122 @@
+package cockroach
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	scdmodels "github.com/interuss/dss/pkg/scd/models"
+	dsssql "github.com/interuss/dss/pkg/sql"
+	"github.com/interuss/stacktrace"
+	"strings"
+	"time"
+)
+
+var (
+	availabilityFieldsWithIndices   [3]string
+	availabilityFieldsWithPrefix    string
+	availabilityFieldsWithoutPrefix string
+)
+
+// TODO Update database schema and fields below.
+func init() {
+	availabilityFieldsWithIndices[0] = "id"
+	availabilityFieldsWithIndices[1] = "availability"
+	availabilityFieldsWithIndices[2] = "updated_at"
+
+	availabilityFieldsWithoutPrefix = strings.Join(
+		availabilityFieldsWithIndices[:], ",",
+	)
+
+	withPrefix := make([]string, len(availabilityFieldsWithIndices))
+	for idx, field := range availabilityFieldsWithIndices {
+		withPrefix[idx] = "scd_uss_availability." + field
+	}
+
+	availabilityFieldsWithPrefix = strings.Join(
+		withPrefix[:], ",",
+	)
+}
+
+// Implements repos.UssAvailability.UpsertAvailability
+func (u *repo) UpsertUssAvailability(ctx context.Context, s *scdmodels.UssAvailabilityStatus) (*scdmodels.UssAvailabilityStatus, error) {
+	var (
+		upsertQuery = fmt.Sprintf(`
+		UPSERT INTO
+		scd_uss_availability
+		  (%s)
+		VALUES
+			($1, $2, transaction_timestamp())
+		RETURNING
+			%s`, availabilityFieldsWithoutPrefix, availabilityFieldsWithPrefix)
+	)
+
+	s, err := u.fetchAvailability(ctx, u.q, upsertQuery,
+		s.Uss,
+		s.Availability)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "Error fetching Availability")
+	}
+	return s, nil
+}
+
+func (u *repo) fetchAvailabilities(ctx context.Context, q dsssql.Queryable, query string, args ...interface{}) ([]*scdmodels.UssAvailabilityStatus, error) {
+	rows, err := q.QueryContext(ctx, query, args...)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "Error in query: %s", query)
+	}
+	defer rows.Close()
+
+	var payload []*scdmodels.UssAvailabilityStatus
+	for rows.Next() {
+		var (
+			u         = new(scdmodels.UssAvailabilityStatus)
+			updatedAt time.Time
+		)
+		err := rows.Scan(
+			&u.Uss,
+			&u.Availability,
+			&updatedAt,
```

Comment:
It looks like the timestamp is scanned, but then we don't do anything with it.  How will `UssAvailabilityStatus.Version` get populated when reading from the database?